mava/networks/mat_network.py

```diff
@@ -0,0 +1,274 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Tuple
+
+import chex
+import jax.numpy as jnp
+from flax import linen as nn
+from flax.linen.initializers import orthogonal
+
+from mava.networks.attention import SelfAttention
+from mava.networks.torsos import SwiGLU
+from mava.networks.utils.mat.decode import (
+    continuous_autoregressive_act,
+    continuous_parallel_act,
+    discrete_autoregressive_act,
+    discrete_parallel_act,
+)
+from mava.systems.mat.types import MATNetworkConfig
+from mava.types import MavaObservation
+from mava.utils.network_utils import _CONTINUOUS, _DISCRETE
```

Comment:
NIT:
```suggestion
from mava.utils.network_utils import CONTINUOUS, DISCRETE
```

Reply:
This import won't work since the variable are called `_CONTINUOUS` and `_DISCRETE` in `mava.utils.network_utils`.

Reply:
Yeah I was thinking of changing that but nvm 😅